src/ReflectionEngine.php

```diff
@@ -239,7 +239,7 @@ public static function parseFileNamespace($fileName, $namespaceName)
                 continue;
             }
             $topLevelNodeName = $topLevelNode->name ? $topLevelNode->name->toString() : '';
-            if ($topLevelNodeName === $namespaceName) {
+            if ($topLevelNodeName === $namespaceName || "\\{$topLevelNodeName}" === $namespaceName) {
```

Comment:
It will be better to normalize this value before any additional actions in the method beginning. Eg.

```php
$namespaceName = ltrim($namespaceName, '\\');
```

Then we will reduce method complexity by avoiding extra check in the if expression.